src/main/java/com/CezaryZal/controller/UserControllerImp.java

```diff
@@ -2,32 +2,37 @@
 
 import com.CezaryZal.entity.User;
 import com.CezaryZal.manager.db.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.Optional;
+import javax.naming.NameNotFoundException;
+import java.util.List;
 
 @RestController
 @RequestMapping("/admin/user")
 public class UserControllerImp implements UserController {
 
     private UserService userService;
 
+    @Autowired
     public UserControllerImp(UserService userService) {
         this.userService = userService;
     }
 
     @GetMapping("/{index}")
-    public Optional<User> findById(@PathVariable Long index) {
-        return userService.findById(index);
+    public ResponseEntity<User> findById(@PathVariable Long index) throws NameNotFoundException {
+        return new ResponseEntity<>(userService.findById(index), HttpStatus.OK);
     }
 
     @GetMapping("/name/{loginName}")
-    public Optional<User> findByLoginName(@PathVariable String loginName){
-        return userService.findByLoginName(loginName);
+    public ResponseEntity<User> findByLoginName(@PathVariable String loginName) throws NameNotFoundException {
```

Comment:
zastanów się czy login w pathvariable jest dobrym pomysłem?
czy jakieś nietypowe znaki w loginie @#$%^&* nie będą problematyczne?

Reply:
A czemu miały by być problematyczne skoro przesyłany jest łańcuch znaków, czy to znaki czy symbole dla aplikacji nie będę problemem